book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -600,20 +600,39 @@ order on byte strings is the numeric ordering).
 
 We use the following rocksdb column families:
 
-| Tree                 |                  Keys |                              Values |
-|----------------------|-----------------------|-------------------------------------|
-| `hash_by_height`     | `BE32(height)`        | `block::Hash`                       |
-| `height_by_hash`     | `block::Hash`         | `BE32(height)`                      |
-| `block_by_height`    | `BE32(height)`        | `Block`                             |
-| `tx_by_hash`         | `transaction::Hash`   | `(BE32(height) \|\| BE32(tx_index))`|
-| `utxo_by_outpoint`   | `OutPoint`            | `TransparentOutput`                 |
-| `sprout_nullifiers`  | `sprout::Nullifier`   | `()`                                |
-| `sapling_nullifiers` | `sapling::Nullifier`  | `()`                                |
-| `orchard_nullifiers` | `orchard::Nullifier`  | `()`                                |
-| `sprout_anchors`     | `sprout::tree::Root`  | `()`                                |
-| `sapling_anchors`    | `sapling::tree::Root` | `()`                                |
+| Tree                  | Keys                  | Values                               | Merge  |
+|-----------------------|-----------------------|--------------------------------------|--------|
+| `hash_by_height`      | `BE32(height)`        | `block::Hash`                        | Never  |
+| `height_by_hash`      | `block::Hash`         | `BE32(height)`                       | Never  |
+| `block_by_height`     | `BE32(height)`        | `Block`                              | Never  |
+| `tx_by_hash`          | `transaction::Hash`   | `(BE32(height) \|\| BE32(tx_index))` | Never  |
+| `utxo_by_outpoint`    | `OutPoint`            | `transparent::Output`                | Delete |
+| `sprout_nullifiers`   | `sprout::Nullifier`   | `()`                                 | Never  |
+| `sapling_nullifiers`  | `sapling::Nullifier`  | `()`                                 | Never  |
+| `orchard_nullifiers`  | `orchard::Nullifier`  | `()`                                 | Never  |
+| `sprout_anchors`      | `sprout::tree::Root`  | `()`                                 | Never  |
+| `sprout_incremental`  | *TODO*                | *TODO*                               | *TODO* |
+| `sapling_anchors`     | `sapling::tree::Root` | `()`                                 | Never  |
+| `sapling_incremental` | *TODO*                | *TODO*                               | *TODO* |
+| `orchard_anchors`     | `orchard::tree::Root` | `()`                                 | Never  |
+| `orchard_incremental` | *TODO*                | *TODO*                               | *TODO* |
```

Comment:
@dconnolly will we ever want to insert the same anchor twice?
What happens if there are no transactions for a particular pool in a block?

Either way is fine, I just need to update the design here.

Reply:
Overwrites can happen.